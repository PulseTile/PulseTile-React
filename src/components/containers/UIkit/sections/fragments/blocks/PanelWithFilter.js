import React, { Component } from 'react';

const panelFilterClassNameDefault = 'panel-filter';

/**
 * This component returns content of PanelWithFilter section in Blocks
 *
 * @return {XML}
 * @constructor
 */
export default class PanelWithFilter extends Component {

  state = {
    panelFilterClassName: 'panel-filter',
    isFilterOpen: false,
  };

  /**
   * This function hide/show filter input
   */
  toggleFilter = () => {
    const { isFilterOpen } = this.state;
    if (isFilterOpen) {
      this.setState({
        panelFilterClassName: panelFilterClassNameDefault,
        isFilterOpen: false,
      });
    } else {
      this.setState({
        panelFilterClassName: panelFilterClassNameDefault + ' hidden',
        isFilterOpen: true,
      });
    }
  };

  render() {
    const { panelFilterClassName } = this.state;
    return (
      <div id="panel-filter" className="ui-section">
        <strong className="ui-title">Panel heading with filter</strong>
        <div className="ui-sub-section">
          <div className="panel panel-primary">
            <div className="panel-heading">
              <div className="control-group right">
                <button type="button" className="btn btn-success btn-inverse btn-filter" onClick={() => this.toggleFilter()}>
                  <i className="btn-icon fa fa-filter"></i>
                </button>
              </div>
              <h3 className="panel-title">Panel title</h3>
              <div className={panelFilterClassName}>
                <div className="inner-addon addon-left">
                  <div className="addon">
                    <i className="fa fa-filter"></i>
                  </div>
                  <input type="text" id="filter" className="form-control" placeholder="Filter..." />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
