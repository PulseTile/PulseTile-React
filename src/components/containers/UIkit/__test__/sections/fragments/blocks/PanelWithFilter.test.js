import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import PanelWithFilter from '../../../../sections/fragments/blocks/PanelWithFilter';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <PanelWithFilter />', () => {
  it('should renders PanelWithFilter with props correctly', () => {
    const component = shallow(<PanelWithFilter />);
    component.find('.btn.btn-success.btn-inverse.btn-filter').simulate('click');
    expect(component).toMatchSnapshot();
    component.find('.btn.btn-success.btn-inverse.btn-filter').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
