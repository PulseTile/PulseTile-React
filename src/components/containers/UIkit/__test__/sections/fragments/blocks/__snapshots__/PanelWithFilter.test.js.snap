// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PanelWithFilter /> should renders PanelWithFilter with props correctly 1`] = `
<div
  className="ui-section"
  id="panel-filter"
>
  <strong
    className="ui-title"
  >
    Panel heading with filter
  </strong>
  <div
    className="ui-sub-section"
  >
    <div
      className="panel panel-primary"
    >
      <div
        className="panel-heading"
      >
        <div
          className="control-group right"
        >
          <button
            className="btn btn-success btn-inverse btn-filter"
            onClick={[Function]}
            type="button"
          >
            <i
              className="btn-icon fa fa-filter"
            />
          </button>
        </div>
        <h3
          className="panel-title"
        >
          Panel title
        </h3>
        <div
          className="panel-filter hidden"
        >
          <div
            className="inner-addon addon-left"
          >
            <div
              className="addon"
            >
              <i
                className="fa fa-filter"
              />
            </div>
            <input
              className="form-control"
              id="filter"
              placeholder="Filter..."
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <PanelWithFilter /> should renders PanelWithFilter with props correctly 2`] = `
<div
  className="ui-section"
  id="panel-filter"
>
  <strong
    className="ui-title"
  >
    Panel heading with filter
  </strong>
  <div
    className="ui-sub-section"
  >
    <div
      className="panel panel-primary"
    >
      <div
        className="panel-heading"
      >
        <div
          className="control-group right"
        >
          <button
            className="btn btn-success btn-inverse btn-filter"
            onClick={[Function]}
            type="button"
          >
            <i
              className="btn-icon fa fa-filter"
            />
          </button>
        </div>
        <h3
          className="panel-title"
        >
          Panel title
        </h3>
        <div
          className="panel-filter"
        >
          <div
            className="inner-addon addon-left"
          >
            <div
              className="addon"
            >
              <i
                className="fa fa-filter"
              />
            </div>
            <input
              className="form-control"
              id="filter"
              placeholder="Filter..."
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
